Error on startup after updating to 3.6.0

### Expected Behavior

The app should continue to run

### Actual Behaviour

When using `3.6.0` the startup fails with

```
Message: No bean of type [io.micronaut.transaction.interceptor.CoroutineTxHelper] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: AsyncTransactionOperations.buildPrimaryAsyncTransactionOperations(SynchronousTransactionManager synchronousTransactionManager,CoroutineTxHelper coroutineTxHelper) --> AsyncTransactionOperations.buildPrimaryAsyncTransactionOperations(SynchronousTransactionManager synchronousTransactionManager,[CoroutineTxHelper coroutineTxHelper])
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2087)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1297)
	at io.micronaut.data.hibernate.transaction.$HibernateTransactionManagersFactory$BuildPrimaryAsyncTransactionOperations0$Definition.doBuild(Unknown Source)
	at io.micronaut.context.AbstractInitializableBeanDefinition.build(AbstractInitializableBeanDefinition.java:769)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:158)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2354)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2918)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2737)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1915)
	... 7 common frames omitted
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.transaction.interceptor.CoroutineTxHelper] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2805)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1617)
	at io.micronaut.context.AbstractBeanResolutionContext.getBean(AbstractBeanResolutionContext.java:66)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2065)
	... 19 common frames omitted

```

### Steps To Reproduce

```groovy
plugins {
	id("com.github.johnrengelman.shadow") version "7.1.2"
	id("io.micronaut.application") version "3.5.1"
	id "com.gorylenko.gradle-git-properties" version "2.3.1"
}

dependencies {
	implementation "io.micronaut:micronaut-runtime"
	implementation 'io.projectreactor:reactor-core'

	annotationProcessor 'io.micronaut.data:micronaut-data-processor'
	implementation 'io.micronaut.data:micronaut-data-hibernate-jpa'
	implementation 'org.postgresql:postgresql:42.3.6'
	implementation "io.micronaut.flyway:micronaut-flyway"
	implementation "io.micronaut.sql:micronaut-jdbc-hikari"
	implementation "jakarta.annotation:jakarta.annotation-api"
	implementation "io.micronaut.sql:micronaut-hibernate-jpa"
	implementation 'com.vladmihalcea:hibernate-types-52:2.17.1'
	implementation "io.micronaut:micronaut-validation"
	implementation "io.micronaut.security:micronaut-security"
	implementation "io.micronaut.security:micronaut-security-annotations"
	annotationProcessor "io.micronaut.openapi:micronaut-openapi"
	implementation "io.swagger.core.v3:swagger-annotations"
	testImplementation 'io.micronaut:micronaut-http-client'
}

micronaut {
	version '3.6.0'
	runtime("netty")
	testRuntime("junit5")
	processing {
		incremental(true)
		annotations("com.company.*")
	}
}

```

### Environment Information

- OS: Windows 11
- JDK: Eclipse Temurin 17.0.1

### Example Application

_No response_

### Version

3.6.0

--- a/data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/transaction/HibernateTransactionManagersFactory.java
+++ b/data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/transaction/HibernateTransactionManagersFactory.java
@@ -19,6 +19,7 @@ import io.micronaut.context.annotation.EachBean;
 import io.micronaut.context.annotation.Factory;
 import io.micronaut.context.annotation.Parameter;
 import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.Nullable;
 import io.micronaut.transaction.SynchronousTransactionManager;
 import io.micronaut.transaction.async.AsyncTransactionOperations;
 import io.micronaut.transaction.async.AsyncUsingSyncTransactionOperations;
@@ -38,7 +39,7 @@ final class HibernateTransactionManagersFactory {
 
     @EachBean(DataSource.class)
     <T> AsyncTransactionOperations<T> buildPrimaryAsyncTransactionOperations(@Parameter SynchronousTransactionManager<T> synchronousTransactionManager,
-                                                                             CoroutineTxHelper coroutineTxHelper) {
+                                                                             @Nullable CoroutineTxHelper coroutineTxHelper) {
         return new AsyncUsingSyncTransactionOperations<>(synchronousTransactionManager, coroutineTxHelper);
     }
 

